src/inventory_ui.cpp

```diff
@@ -2908,3 +2917,79 @@ inventory_selector::stats inventory_drop_selector::get_raw_stats() const
                u.volume_capacity_with_tweaks( to_use ),
                u.max_single_item_length(), u.max_single_item_volume() );
 }
+
+void inventory_examiner::action_examine( const item_location sitem, const bool fit_to_window )
+{
+    if( !fit_to_window ) {
+        inventory_selector::action_examine( sitem );
+    } else {
+        // Code below pulled from the action_examine function in advanced_inv.cpp
+        std::vector<iteminfo> vThisItem;
+        std::vector<iteminfo> vDummy;
+
+        sitem->info( true, vThisItem );
+
+        item_info_data data( sitem->tname(), sitem->type_name(), vThisItem, vDummy, examine_window_scroll );
+        //Tries to create a window covering the right half of the inventory screen
+        data.without_getch = true;
+
+        // NOTE: This assumes that the first column (currently own_inv_column) is used for the item list:
+        int inv_column_width = get_all_columns().front()->get_width();
+        /* NOTE: By flagging force_max_window_size as true, this forces the inventory_selector window to be
+         * TERMX x TERMY , allowing us to assume the window starts at 0,0 and is TERMX by TERMY.  If this is
+         * changed in inventory_selector::prepare_layout, then everything breaks.  This obviously isn't a
+         * great solution */
+
+        int border_width = 1; //Assumed, since the value in inventory_selector is private, but reasonable
+        int width = TERMX - 2 * border_width  - inv_column_width;
+        int height = TERMY  - get_header_height() - 1;
+        point start_position = point( ( inv_column_width + border_width + 1 ),
+                                      get_header_height() + 1 );
+
+        draw_item_info( catacurses::newwin( height, width, start_position ), data );
```

Comment:
This creates and destroys the info window every draw cycle, which seems excessive though it simplifies resizing. You could create this window in a dummy `ui_adaptor::on_screen_resize()` since `prepare_layout()` and others are private.

<details>

<summary>quick demo patch</summary>

```diff
diff --git a/src/inventory_ui.cpp b/src/inventory_ui.cpp
index b3e447b0ce..7b79e4ada2 100644
--- a/src/inventory_ui.cpp
+++ b/src/inventory_ui.cpp
@@ -2933,28 +2933,36 @@ void inventory_examiner::action_examine( const item_location sitem, const bool f
         //Tries to create a window covering the right half of the inventory screen
         data.without_getch = true;
 
-        // NOTE: This assumes that the first column (currently own_inv_column) is used for the item list:
-        int inv_column_width = get_all_columns().front()->get_width();
-        /* NOTE: By flagging force_max_window_size as true, this forces the inventory_selector window to be
-         * TERMX x TERMY , allowing us to assume the window starts at 0,0 and is TERMX by TERMY.  If this is
-         * changed in inventory_selector::prepare_layout, then everything breaks.  This obviously isn't a
-         * great solution */
-
-        int border_width = 1; //Assumed, since the value in inventory_selector is private, but reasonable
-        int width = TERMX - 2 * border_width  - inv_column_width;
-        int height = TERMY  - get_header_height() - 1;
-        point start_position = point( ( inv_column_width + border_width + 1 ),
-                                      get_header_height() + 1 );
-
-        draw_item_info( catacurses::newwin( height, width, start_position ), data );
+        draw_item_info( w_ex, data );
     }
 }
 
 item_location inventory_examiner::execute()
 {
     shared_ptr_fast<ui_adaptor> ui = create_or_get_ui_adaptor();
+    ui_adaptor dummy;
+    dummy.on_screen_resize( [this]( ui_adaptor &ui) {
+        // NOTE: This assumes that the first column (currently own_inv_column) is used for the item
+        // list:
+        int inv_column_width = get_all_columns().front()->get_width();
+        /* NOTE: By flagging force_max_window_size as true, this forces the inventory_selector
+         * window to be TERMX x TERMY , allowing us to assume the window starts at 0,0 and is TERMX
+         * by TERMY.  If this is changed in inventory_selector::prepare_layout, then everything
+         * breaks.  This obviously isn't a great solution */
+
+        int border_width =
+            1; // Assumed, since the value in inventory_selector is private, but reasonable
+        int width = TERMX - 2 * border_width - inv_column_width;
+        int height = TERMY - get_header_height() - 1;
+        point start_position =
+            point( ( inv_column_width + border_width + 1 ), get_header_height() + 1 );
+        w_ex = catacurses::newwin( height, width, start_position );
+        ui.position_from_window( w_ex );
+    } );
+    dummy.mark_resize();
     while( true ) {
-        ui_manager::redraw();
+        ui->invalidate_ui();
+        ui_manager::redraw_invalidated();
 
         const inventory_entry &selected = get_active_column().get_selected();
         if( selected ) {
diff --git a/src/inventory_ui.h b/src/inventory_ui.h
index 02ace17a7b..e03f6b14af 100644
--- a/src/inventory_ui.h
+++ b/src/inventory_ui.h
@@ -833,6 +833,7 @@ class inventory_examiner : public inventory_selector
 {
     private:
         int examine_window_scroll;
+        catacurses::window w_ex;
     public:
         explicit inventory_examiner( Character &p,
                                      const inventory_selector_preset &preset = default_preset ) :
```
</details>

You can also add `action_examine()` to this callback to solve your resizing quirk. Not sure if this is appropriate style though; maybe @Qrox can weigh in.

Reply:
I think you can just override the `create_or_get_ui_adaptor` function and move the item info window creation and drawing code to the callbacks there. Or, just pretend the item info window is an overlay instead of a part of the inventory menu, and create a separate `ui_adaptor` for it. Do note that the game window may resize while `get_input` is blocking for input, so redrawing it after `ui_manager::redraw` instead of in a redraw callback will make it resize incorrectly.

Reply:
I'm not terribly familiar with ui_adaptor right now, but I'll do some playing around with it.  Right now I'd favour treating it as an overlay rather than trying to mess with any underlying functions.

How significant an impact is it to create the window every cycle?  Given that it's waiting for user input every time I'd file it under "Not a great solution.  Kind of annoying", but are there are any things happening further under the hood that might make it more of an issue?